component/azstorage/datalake.go

```diff
@@ -578,6 +605,18 @@ func (dl *Datalake) TruncateFile(name string, size int64) error {
 	return dl.BlockBlob.TruncateFile(name, size)
 }
 
+func (dl *Datalake) getACL(name string) (string, error) {
+	log.Trace("Datalake::getACL : Get ACLs for file %s", name)
+	fileClient := dl.Filesystem.NewFileClient(filepath.Join(dl.Config.prefixPath, name))
+	acl, err := fileClient.GetAccessControl(context.Background(), nil)
+	if err != nil {
+		log.Err("Datalake::getACL : Failed to get ACLs for file %s [%s]", name, err.Error())
+		return "", err
+	}
+
+	return *acl.ACL, nil
```

Comment:
This can result in panic in these cases,
1. Error is not nil and acl is nil
2. Error is nil, acl is not nil, but acl.ACL is nil

You can have a check something like,
`if acl != nil && acl.ACL != nil`

Reply:
This method can be removed as its needed only by test and not used in datalake.go code anymore. Will try to refactor the code here.